fs-sim/fs-sim.cabal

```diff
@@ -59,9 +59,13 @@ test-suite fs-sim-test
   hs-source-dirs:   test src
   main-is:          Main.hs
   other-modules:
+    System.FS.Sim.Error
     System.FS.Sim.FsTree
     System.FS.Sim.MockFS
     System.FS.Sim.Prim
+    System.FS.Sim.STM
+    System.FS.Sim.Stream
```

Comment:
Is there a particular reason why the test-suite declares the modules directly instead of pulling them from the library?

Reply:
It's a side effect of not exposing `System.FS.Sim.Prim` as a public module. The test suite therefore does not depend on `fs-sim`, it compiles the files itself

I actually like this alternative approach better: have a private sub-library that contains everything you don't want to expose, and then expose what you need from the main (public) library. However, that would require more refactoring of its own so this is the current intermediate solution that works adequately

Reply:
Then this stop-gap looks good. I would be in favor of the alternative proposal though